geode-core/src/main/java/org/apache/geode/internal/cache/DirectoryHolder.java

```diff
@@ -44,15 +47,15 @@
 
   /** For testing purposes we can set the disk directory size in bytes **/
   @MutableForTesting
-  static boolean SET_DIRECTORY_SIZE_IN_BYTES_FOR_TESTING_PURPOSES = false;
+  static DiskDirSizesUnit DISK_DIR_SIZES_UNIT = DiskDirSizesUnit.MEGABYTES;
 
   DirectoryHolder(StatisticsFactory factory, File dir, long space, int index) {
     this(dir.getPath(), factory, dir, space, index);
   }
 
   DirectoryHolder(String ownersName, StatisticsFactory factory, File dir, long space, int index) {
     this.dir = dir;
-    if (SET_DIRECTORY_SIZE_IN_BYTES_FOR_TESTING_PURPOSES) {
+    if (DISK_DIR_SIZES_UNIT == DiskDirSizesUnit.BYTES) {
```

Comment:
If the disk unit is going to be an `enum` instead of a `bool`, consider checking each case explicitly in an `else if` or `switch` and then throwing an exception if the value doesn't match any of the options. This prevents accidentally falling through to the default case if we ever add a new `enum` value.
```suggestion
    if (DISK_DIR_SIZES_UNIT == DiskDirSizesUnit.BYTES) {
      this.capacity = space;
    } else if (DISK_DIR_SIZES_UNIT == DiskDirSizesUnit.MEGABYTES) {
      // convert megabytes to bytes
      this.capacity = space * 1024 * 1024;
    } else {
      throw new SomeException(...)
    }
```